ludwig/models/predictor.py

```diff
@@ -264,10 +264,7 @@ def calculate_overall_stats(
 ):
     overall_stats = {}
     for of_name, output_feature in output_features.items():
-        if of_name not in overall_stats:
-            overall_stats[of_name] = {}
-        output_feature.calculate_overall_stats(
-            overall_stats[of_name], # dictionary for overall stats
+        overall_stats[of_name] = output_feature.calculate_overall_stats(
```

Comment:
I believe this is the only usage of `calculate_overall_stats`.  As such, my feeling is we should make this a non-static method.

Reply:
Originally when I made these dat aprocessing functions static (both preprocessing and postprocessing) the idea is that the fact that they were stateless would have made it easier in the future to run them in parallel. Now I'm not 100% sure, making them non static will likely make for more readable and easier code. I'm on the fence on this.